### ISSUE TITLE
Running with some prombles

### ISSUE THREAD

**Issue by NiceDay-gogogo** [yriyazi/High-Resolution-Remote-Sensing-Image-Captioning-Based-on-Structured-Attention-and-SAM-Network#3]
I have posted my steps and questions at the following link. Can you see them? I look forward to your reply. Thank you very much.

https://blog.csdn.net/weixin_48782031/article/details/145367575?sharetype=blogdetail&sharerId=145367575&sharerefer=PC&sharesource=weixin_48782031&spm=1011.2480.3001.8118

**Comment by yriyazi** [comment]
Dear zhantianle,
If I don't remember the accuracy of the train. However, you can use the
test.py and check the BLEU score of the test data.
I remember we used the UCM dataset. Are you using UCM as well?
If you use the default value you should be able to achieve the following
result. I have included it in our paper.
The parameters that you have changed have an effect on the speed of
convergence and I believe 20 epochs is enough.
So for further diagnosis please include the BLEU score 1 to 4 as well
Best regards,
Yassin Riyazi

[image: image.png]

**Comment by NiceDay-gogogo** [comment]
The results are shown in the figure.Is this correct? iteration : 7: 100%|█| 7/7 [01:40<00:00, 14.29s/it, BLEU_1=0.6388, BLEU_2=0.5391
[0.63878951 0.5391072  0.47078808 0.42256291]

**Comment by NiceDay-gogogo** [comment]
I made additional modifications and published the steps at the following link.  looking forward to your reply. Thankyou！
https://blog.csdn.net/weixin_48782031/article/details/145379595?sharetype=blogdetail&sharerId=145379595&sharerefer=PC&sharesource=weixin_48782031&spm=1011.2480.3001.8118

**Comment by yriyazi** [comment]
Dear zhantianle,
I hope you are doing well.

The training process is going rather smoothly, however, I don't
remember the exact values for each hyper parameter.
You can refer to the ICEE paper
<https://ieeexplore.ieee.org/document/10668247> or use the hyperparameter
value.
We have done an ablation study on a different combination of losses and
effects of warm-up over neural network performance.
Or you can use the default "config.yaml" values.

I am not sure that you did calculate the BLEU score for all test samples.
it should be calculated for all samples and then calculate the average.

I recommend experimenting with the learning rate to get a better validation
accuracy and then calculating the BLEU score.
Because it is so computationally intensive.

Best regards,
Yassin Riyazi
